lib/android.js

```diff
@@ -1882,7 +1885,13 @@ function ensureArtKnowsHowToHandleReplacementMethods (vm) {
   const apiLevel = getAndroidApiLevel();
 
   let exportName = null;
-  if (apiLevel > 28) {
+  
+  const api = getApi();
+  const kCollectorTypeCMC = 3;
+  let mayUseCollector = new NativeFunction(Module.findExportByName('libart.so', "_ZNK3art2gc4Heap15MayUseCollectorENS0_13CollectorTypeE"), "int", ["pointer", "int"])
```

Comment:
I checked libart back to nougat and the method mayUseCollector was present at least since then.
I completely agree that hooking both the MarkCompact and the ConcurrentCopying is better avoiding the check on the current GC.
I can change it